homeassistant/components/sky_hub/device_tracker.py

```diff
@@ -1,111 +1,62 @@
 """Support for Sky Hub."""
 import logging
-import re
 
-import requests
+from pyskyqhub.skyq_hub import SkyQHub
 import voluptuous as vol
 
 from homeassistant.components.device_tracker import (
     DOMAIN,
     PLATFORM_SCHEMA,
     DeviceScanner,
 )
-from homeassistant.const import CONF_HOST, HTTP_OK
+from homeassistant.const import CONF_HOST
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
 import homeassistant.helpers.config_validation as cv
 
 _LOGGER = logging.getLogger(__name__)
-_MAC_REGEX = re.compile(r"(([0-9A-Fa-f]{1,2}\:){5}[0-9A-Fa-f]{1,2})")
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({vol.Optional(CONF_HOST): cv.string})
 
 
-def get_scanner(hass, config):
+async def async_get_scanner(hass, config):
     """Return a Sky Hub scanner if successful."""
-    scanner = SkyHubDeviceScanner(config[DOMAIN])
+    host = config[DOMAIN].get(CONF_HOST, "192.168.1.254")
+    websession = async_get_clientsession(hass)
+    hub = SkyQHub(websession, host)
 
-    return scanner if scanner.success_init else None
+    _LOGGER.info("Initialising Sky Hub")
+    await hub.async_connect()
+    if hub.success_init:
+        scanner = SkyHubDeviceScanner(config[DOMAIN], hub)
+        return scanner
+
+    return None
 
 
 class SkyHubDeviceScanner(DeviceScanner):
     """This class queries a Sky Hub router."""
 
-    def __init__(self, config):
+    def __init__(self, config, hub):
         """Initialise the scanner."""
-        _LOGGER.info("Initialising Sky Hub")
-        self.host = config.get(CONF_HOST, "192.168.1.254")
+        self._hub = hub
         self.last_results = {}
-        self.url = f"http://{self.host}/"
-
-        # Test the router is accessible
-        data = _get_skyhub_data(self.url)
-        self.success_init = data is not None
 
-    def scan_devices(self):
+    async def async_scan_devices(self):
         """Scan for new devices and return a list with found device IDs."""
-        self._update_info()
-
-        return (device for device in self.last_results)
-
-    def get_device_name(self, device):
-        """Return the name of the given device or None if we don't know."""
-        # If not initialised and not already scanned and not found.
-        if device not in self.last_results:
-            self._update_info()
-
-            if not self.last_results:
-                return None
+        await self._async_update_info()
+        return list(device for device in self.last_results)
```

Comment:
```suggestion
        return [device for device in self.last_results]
```

Reply:
With this change I get this pylint error. I didn't get it when I did the list(xxxxx) expression.

`return [device for device in self.last_results]`

```
Run . venv/bin/activate
8
************* Module homeassistant.components.sky_hub.device_tracker
9
##[warning]homeassistant/components/sky_hub/device_tracker.py:47:0: R1721: Unnecessary use of a comprehension (unnecessary-comprehension)
10
##[error]Process completed with exit code 8.
```